File: Examples/test-suite/matlab/director_string_runme.m
import director_string.*


B=@(string) subclass(A(string),'get_first',@director_string_helper_get_first,'process_text',@director_string_helper_process_text);


b = B('hello');

b.get(0);
if (~strcmp(b.get_first(),'hello world!'))
  error(b.get_first())
end

b.call_process_func();

if (~strcmp(b.smem,'hello'))
  error(b.smem)
end

  

File: Examples/test-suite/php/director_string_runme.php
<?php

require "tests.php";

// No new functions
check::functions(array());
// New classes
check::classes(array('A','StringVector'));
// No new vars
check::globals(array());

class B extends A {
  public $smem;

  function get_first() {
    return parent::get_first() . " world!";
  }

  function process_text($string) {
    parent::process_text($string);
    $this->smem = "hello";
  }
}

$b = new B("hello");

check::equal($b->get(0), "hello", "get(0) failed");

check::equal($b->get_first(), "hello world!", "get_first failed");

check::equal($b->call_get_first(), "hello world!", "call_get_first failed");

$b->call_process_func();

check::equal($b->smem, "hello", "smem failed");

check::done();
